### PR TITLE
Split FontPicker into two

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

For making reusing this component easier in #10840 and #10808.

## Summary

* `StoryFontPicker` can be used in multiple places (design panel, floating toolbar)
* `FontPicker` is a generalized font picker that can be used by the `FontCheckDialog`

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

**Comment by BrookeGraham** [comment]
Ah yes! This makes sense. Do we want to approve this and merge it in to unblock myself and @timarney ? Or do you want me to incorporate this into my pr #10840

**Comment by swissspidy** [comment]
I think merging this one first to unblock both PRs would be best. Though the tests need updating here. If you can lend a hand that'd be awesome!

**Comment by BrookeGraham** [comment]
>Though the tests need updating here. If you can lend a hand that'd be awesome!

Yeah. I'll take over this and update those tests this afternoon.

File: packages/story-editor/src/components/storyFontPicker/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          const updates = updateProperties(element, update, commitValues);
+
+          return {
+            ...updates,
+          };
+        },
+      });
+    },
+    [updateSelectedElements]
+  );
```
Comment (BrookeGraham) [inline packages/story-editor/src/components/storyFontPicker/index.js]: Locally I am seeing this get called twice, but in the unit test I am seeing it getting called 6 times. So I will look into why that might be. 

Screen Shot 2022-03-08 at 2 41 30 PM

https://github.com/GoogleForCreators/web-stories-wp/runs/5471047742?check_suite_focus=true#step:6:107

File: packages/story-editor/src/components/fontPicker/index.js [hunk 2]
```diff
@@ -26,50 +26,31 @@ import {
...
-import { MULTIPLE_DISPLAY_VALUE, MULTIPLE_VALUE } from '../../../../constants';
-import { getCommonValue } from '../../shared';
-import useRichTextFormatting from './useRichTextFormatting';
-import getClosestFontWeight from './getClosestFontWeight';
+import { useFont } from '../../app';
+import { MULTIPLE_DISPLAY_VALUE, MULTIPLE_VALUE } from '../../constants';
 
 const FontPicker = forwardRef(function FontPicker(
-  { selectedElements, pushUpdate, highlightStylesOverride },
+  { onChange, currentValue, highlightStylesOverride },
```
Comment (timarney) [inline packages/story-editor/src/components/fontPicker/index.js:37]: Per 

https://github.com/GoogleForCreators/web-stories-wp/pull/10808#discussion_r821501339

I added a `z-Index` prop (if there's a better way to implement this let me know ... should that be added #10808 or in this PR?)

Commit:
https://github.com/GoogleForCreators/web-stories-wp/pull/10808/commits/8bd605b011001c4cb1573af393e5dc525b9919f1#diff-a802a3ecd6f447cf10aa58abf9e7f3d4e7ec9b458ccba02a3b39ad2a7df4f48fR190


👉  `z-Index` issue noted here:
https://github.com/GoogleForCreators/web-stories-wp/issues/10014#issuecomment-1057569298

Comment (swissspidy) [reply packages/story-editor/src/components/fontPicker/index.js:37]: In #10808 should be fine

**Comment by BrookeGraham** [comment]
Sorry for delay. @swissspidy wanna take a peek at this. I think all the test should be passing now.

**Review by BrittanyIRL — APPROVED** [review]
Awesome! this makes a lot of sense.

**Pull request merged**